fedora-builder/src/main/java/org/dataconservancy/nihms/builder/fs/PassJsonFedoraAdapter.java

```diff
@@ -237,6 +254,65 @@ public URI passToFcrepo(HashMap<URI, PassEntity> entities) {
         return submissionUri;
     }
 
+    /**
+     * Resolves the content referenced by {@link File#getUri()}, uploads the binary to Fedora, and then updates the
+     * {@code File uri} with the location of the binary in the repository.
+     *
+     * @param f a File entity that may have a URI that links to binary content
+     * @param client client used to update the File URI in the repository
+     */
+    private void uploadBinary(File f, PassClient client) {
+        // attempt to upload binary content to fedora
+        if (f.getUri() == null) {
+            return;
+        }
+
+        String contentUri = f.getUri().toString();
+
+        Resource contentResource = null;
+        if (contentUri.startsWith("http") || contentUri.startsWith("file:")) {
+            try {
+                contentResource = new UrlResource(f.getUri());
+            } catch (MalformedURLException e) {
+                throw new RuntimeException(e.getMessage(), e);
+            }
+        }
+
+        if (contentUri.startsWith("classpath*:")) {
```

Comment:
@Trumbore I think that because each `if` is testing a distinct condition, at most one of those three blocks will ever be executed, so I think it's ok, unless you see a bug lurking!

Reply:
@Trumbore Ah but I do see a bug, one moment